test/integration/targets/vmware_guest_controller/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+# Test code for the vmware_guest_controller module
+# Copyright: (c) 2019, Diane Wang (Tomorrow9) <dianew@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- when: vcsim is not defined
+  block:
+  - import_role:
+      name: prepare_vmware_tests
+    vars:
+      setup_attach_host: true
+      setup_datastore: true
+      setup_virtualmachines: true
+
+  - name: gather disk controllers facts of the virtual machine
+    vmware_guest_controller:
+      validate_certs: False
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      datacenter: "{{ dc1 }}"
+      name: "{{ virtual_machines[0].name }}"
+      gather_disk_controller_facts: true
+    register: disk_controller_facts
+
+  - debug: var=disk_controller_facts
+
+  - name: get the number of existing disk controllers
+    set_fact:
+      scsi_num: "{{ disk_controller_facts.disk_controller_data.scsi | length }}"
+  - set_fact:
+      sata_num: "{{ disk_controller_facts.disk_controller_data.sata | length }}"
+  - set_fact:
+      nvme_num: "{{ disk_controller_facts.disk_controller_data.nvme | length }}"
+
+  - name: add new disk controllers to virtual machine
+    vmware_guest_controller:
+      validate_certs: False
+      hostname: "{{ vcenter_hostname }}"
+      username: "{{ vcenter_username }}"
+      password: "{{ vcenter_password }}"
+      datacenter: "{{ dc1 }}"
+      name: "{{ virtual_machines[0].name }}"
+      disk_controller:
```

Comment:
TASK [vmware_guest_controller : add new disk controllers to virtual machine] *****************************************************************************************************************
The step fails with the following error:
```
fatal: [testhost]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (vmware_guest_controller) module: disk_controller Supported parameters include: controllers, datacenter, folder, gather_disk_controller_facts, hostname, moid, name, password, port, proxy_host, proxy_port, sleep_time, use_instance_uuid, username, uuid, validate_certs"}
```

Reply:
Thanks for testing, my mistake not changing parameter in test case. Thanks.